### DISCUSSION TITLE
FullyFusedMLP is not supported for the selected architecture 70. Falling back to CutlassMLP

### DISCUSSION THREAD

**Discussion by ricshaw** [NVlabs/instant-ngp#553] [Q&A]
Following #219 I get this error using headless:
Warning: FullyFusedMLP is not supported for the selected architecture 70. Falling back to CutlassMLP. For maximum performance, raise the target GPU architecture to 75+.

It still manages to train, but this is not ideal.
I built it with TCNN_CUDA_ARCHITECTURES=70 cmake ..

gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)
Cuda compilation tools, release 10.2, V10.2.89
cmake version 3.22.2
V100 GPU

**Comment by Tom94** [comment]
Hi, yes, the V100 GPU is not supported by `FullyFusedMLP`. I'm afraid the only way to benefit from faster performance in instant-ngp will be to upgrade.

**Reply by ghost** [reply]
What about the NVIDIA Titan Xp? Get the same error like above, even when changing to the correct architecture. :(

**Reply by Tom94** [reply]
The Titan Xp is also not supported. Any GPU before Turing (architecture lower than 75) can't run `FullyFusedMLP`.

**Reply by Komti13** [reply]
Hello I also got this error segmentation fault shoul i change the GPU architectude to 75 and how? thanks


22:44:06 SUCCESS  Loaded mesh "data/sdf/bunny.obj" file with 1 shapes.
22:44:06 SUCCESS  Built TriangleBvh: nodes=1365
22:44:06 WARNING  OptiX was not built. Falling back to slower TriangleBVH::ray_intersect.
22:44:06 SUCCESS  Built TriangleOctree: depth=10 nodes=287672 dual_nodes=1150344. Populating dual nodes...
22:44:07 SUCCESS  Loaded mesh: triangles=4968 AABB=[min=[-0.0956273,0.032063,-0.0629261], max=[0.0620257,0.188243,0.0599615]] after scaling=[min=[0.00790903,0.0125811,0.118168], max=[0.992091,0.987419,0.881832]]
22:44:07 INFO     Loading network config from: configs/sdf/base.json
22:44:07 INFO     GridEncoding:  Nmin=16 b=1.38191 F=2 T=2^19 L=16
Warning: FullyFusedMLP is not supported for the selected architecture 61. Falling back to CutlassMLP. For maximum performance, raise the target GPU architecture to 75+.
Warning: FullyFusedMLP is not supported for the selected architecture 61. Falling back to CutlassMLP. For maximum performance, raise the target GPU architecture to 75+.
22:44:07 INFO     Model:         3--[HashGrid]-->32--[FullyFusedMLP(neurons=64,layers=4)]-->1
22:44:07 INFO       total_encoding_params=12196240 total_network_params=6656
Segmentation fault (core dumped)

**Comment by anonymousIoTresearcher** [comment]
Has this issue been resolved? Does TCNN support SM_70 (V100) now?
